cli/src/test/java/uk/ac/ebi/atlas/cli/experiment/CreateUpdateExperimentCommandTest.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    @Test
+    public void whenNoAccessionPassed_ThenReturnCorrectExitCode() {
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(gxaExperimentCrudSpy);
+        var exitCode = createUpdateExperimentCommand.call();
+
+        final int exitCodeForNoExperimentAccessionSupplied = 1;
+        assertThat(exitCode).isEqualTo(exitCodeForNoExperimentAccessionSupplied);
+    }
+
+    @Test
+    public void createExperimentHasBeenCalledByNumberOfProvidedAccessions() {
+        GxaExperimentCrud spiedGxaExperimentCrud = spy(gxaExperimentCrudSpy);
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(spiedGxaExperimentCrud);
+        final UUID uuid1 = randomUUIDs.get(0);
+        final UUID uuid2 = randomUUIDs.get(1);
+        final UUID uuid3 = randomUUIDs.get(2);
+
+        doReturn(uuid1).doReturn(uuid2).doReturn(uuid3).
+                when(spiedGxaExperimentCrud).createExperiment(anyString(),eq(false));
+
+        CommandLine cmd = new CommandLine(createUpdateExperimentCommand);
+
+        int exitCode = cmd.execute("-e=acc1,acc2,acc3");
+
+        assertThat(exitCode).isEqualTo(0);
+        verify(spiedGxaExperimentCrud, times(3))
+                .createExperiment(anyString(), eq(false));
+    }
+
+    @Test
+    public void ifExperimentCreationFails_ThenRecordedInFailedAccessions() throws Exception{
+        GxaExperimentCrud spiedGxaExperimentCrud = spy(gxaExperimentCrudSpy);
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(spiedGxaExperimentCrud);
+        final UUID uuid1 = randomUUIDs.get(0);
+
+        var acc1 = "acc1";
+        var acc2 = "acc2";
+        var acc3 = "acc3";
+
+        var tempOutputFile = File.createTempFile("tmp", "");
+        tempOutputFile.deleteOnExit();
+        var expectedOutputFile = new File("src/test/resources/expectedErrorOutput.txt");
+
+        doReturn(uuid1).
+                when(spiedGxaExperimentCrud).createExperiment(acc1,false);
+        doThrow(NullPointerException.class).
+                when(spiedGxaExperimentCrud).createExperiment(acc2,false);
+        doThrow(NullPointerException.class).
+                when(spiedGxaExperimentCrud).createExperiment(acc3,false);
+
+        CommandLine cmd = new CommandLine(createUpdateExperimentCommand);
+
+        final String failedAccessionPath = String.format("-f=%s", tempOutputFile.getAbsolutePath());
+        final String experiments = String.format("-e=%s,%s,%s", acc1, acc2, acc3);
+        int exitCode = cmd.execute(failedAccessionPath, experiments);
+
+        assertThat(exitCode).isEqualTo(1);
+        assertThat(tempOutputFile).hasSameContentAs(expectedOutputFile);
+    }
+
+    private List<UUID> generateRandomUUIDs(int numberOfUUIDs) {
+        List<UUID> uuids = new ArrayList<>(numberOfUUIDs);
+        for (int i = 0; i < numberOfUUIDs; i++) {
+            uuids.add(UUID.randomUUID());
+        }
+
+        return uuids;
+    }
```

Comment:
Our style-of-choice is to use the streams API and immutable collections as much as possible. E.g.:
```java
private ImmutableList<UUID> generateRandomUUIDs(int numberOfUUIDs) {
    return IntStream.range(0, numberOfUUIDs)
            .boxed()
            .map(__ -> UUID.randomUUID())
            .collect(toImmutableList());
}
```

Not really important in this particular case since it’s also concise, but I think it’s better to stick to one style to make the cognitive load lighter.

Reply:
ok, I am going to apply that.